File: crates/kittypaw-gui/frontend/src/lib/stores/workspace.ts
import { writable } from 'svelte/store';

export interface FileEntry {
	path: string;
	size: number;
	modified: string;
	is_dir: boolean;
}

export interface FileChange {
	id: string;
	path: string;
	change_type: 'Create' | 'Modify' | 'Delete';
	diff: string;
	new_content: string;
	status: 'Pending' | 'Approved' | 'Rejected' | 'Applied';
}

export interface Workspace {
	id: string;
	name: string;
	root_path: string;
	created_at: string;
}

export const currentWorkspace = writable<Workspace | null>(null);
export const fileTree = writable<FileEntry[]>([]);
export const selectedFile = writable<string | null>(null);
export const fileContent = writable<string>('');
export const pendingChanges = writable<FileChange[]>([]);

File: crates/kittypaw-core/src/workspace.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Workspace {
    pub id: String,
    pub name: String,
    pub root_path: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FileEntry {
    pub path: String, // relative to workspace root
    pub size: u64,
    pub modified: String,
    pub is_dir: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FileChange {
    pub id: String,
    pub workspace_id: String,
    pub path: String,
    pub change_type: ChangeType,
    pub diff: String,
    pub new_content: String,
    pub status: FileChangeStatus,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ChangeType {
    Create,
    Modify,
    Delete,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum FileChangeStatus {
    Pending,
    Approved,
    Rejected,
    Applied,
}
